library/lib.sh

```diff
@@ -384,6 +414,16 @@ rolesDistributeSSHKeys() {
     fi
 }
 
+rolesBuildEtcHosts() {
+    local guests_yml=$1
+    managed_nodes=$(rolesGetManagedNodes "$guests_yml")
+    for managed_node in $managed_nodes; do
+        managed_node_ip=$(yq -r ".\"$managed_node\".\"primary-address\"" "$guests_yml")
+        echo "$managed_node_ip" "$managed_node" >> /etc/hosts
```

Comment:
Is there a chance that `$managed_node_ip` or `$managed_node` could already be in `/etc/hosts`?

Reply:
Actually, testing farm should be able to route hostname to ip without this. I will be able to troubleshoot this next week live and see what's best.

Reply:
This is required because we change hostnames of hosts and then want the new hostnames to be recognizable in the network.
/etc/hosts only contains the default localhost lines otherwise.
Currently, this is done in the general test `tests/prep_managed_node/prep_managed_node.sh`, I can move this to prep_mssql nodes only. I think it's good to make it on all managed nodes though. WDYT?

Reply:
>  Currently, this is done in the general test tests/prep_managed_node/prep_managed_node.sh, I can move this to prep_mssql nodes only. I think it's good to make it on all managed nodes though. WDYT?

Yes.  I can foresee some use cases where managed hosts may need to contact other managed hosts over https or other TLS connections and will need a hostname.